code/modules/organs/subtypes/machine.dm

```diff
@@ -62,26 +62,101 @@
 	desc = "A small, powerful cell for use in fully prosthetic bodies."
 	icon = 'icons/obj/power.dmi'
 	icon_state = "scell"
-	organ_tag = "cell"
+	organ_tag = BP_CELL
 	parent_organ = BP_CHEST
-	vital = 1
-	var/emp_counter = 0
+	vital = TRUE
+	max_damage = 80
+	relative_size = 80
+	var/open = FALSE
+	var/obj/item/cell/cell = /obj/item/cell/super
+	//at 0.8 completely depleted after 60ish minutes of constant walking or 130 minutes of standing still
+	var/servo_cost = 0.8
 
 /obj/item/organ/internal/cell/Initialize()
 	robotize()
+	if(ispath(cell))
+		cell = new cell(src)
 	. = ..()
 
+/obj/item/organ/internal/cell/proc/percent()
+	if(!cell)
+		return 0
+	return get_charge()/cell.maxcharge * 100
+
+/obj/item/organ/internal/cell/proc/get_charge()
+	if(!cell)
+		return 0
+	if(status & ORGAN_DEAD)
+		return 0
+	return round(cell.charge*(1 - damage/max_damage))
+
+/obj/item/organ/internal/cell/proc/checked_use(var/amount)
+	if(!is_usable())
+		return FALSE
+	return cell && cell.checked_use(amount)
+
+/obj/item/organ/internal/cell/proc/use(var/amount)
+	if(!is_usable())
+		return 0
+	return cell && cell.use(amount)
+
 /obj/item/organ/internal/cell/process()
 	..()
-	if(emp_counter)
-		emp_counter--
+	if(!owner)
+		return
+	if(owner.stat == DEAD)	//not a drain anymore
+		return
+	var/cost = get_power_drain()
+	if(world.time - owner.l_move_time < 15)
+		cost *= 2
+	if(!checked_use(cost) && owner.isSynthetic())
+		if(!owner.lying && !owner.buckled)
+			to_chat(owner, "<span class='warning'>You don't have enough energy to function!</span>")
+		owner.Paralyse(3)
+
+/obj/item/organ/internal/cell/proc/get_power_drain()	
+	var/damage_factor = 1 + 10 * damage/max_damage
+	return servo_cost * damage_factor
```

Comment:
We have multiple places where cell damage has an effect.

It changes the charge in the cell in `get_charge` AND the drain on the cell here.

I feel like it should be simplified to just one of these.

Reply:
Sure.